ZenPacks/zenoss/Import4/perf.py

```diff
@@ -7,14 +7,155 @@
 #
 ##############################################################################
 
-import argparse
-from ZenPacks.zenoss.Import4.migration import MigrationBase
+import os
+import sys
+import subprocess
+import re
+import time
+
+from ZenPacks.zenoss.Import4.migration import MigrationBase, ImportError
+
+# some common constants shared among the py and bash scripts
+_check_tag = '[Check] '
+_stderr_tag = '[STDERR] '
+_import_prefix = 'Perf: '
+
+_import4_vol = '/import4'
+_import4_pkg = '/import4/pkg'
+_import4_pkg_bin = '/import4/pkg/bin'
+
+_tasks_Q = '%s/Q.tasks' % _import4_vol
+_jobs_Q = '%s/Q.jobs' % _import4_vol
+_tsdb_Q = '%s/Q.tsdb' % _import4_vol
+
+_jobs_done = '%s/.done' % _jobs_Q
+_tsdb_done = '%s/.done' % _tsdb_Q
+
+
+class Results(object):
+    RUNTIME_ERROR = 'PERF_MIGRATION_RUNTIME_ERROR'
+    COMMAND_ERROR = 'PERF_MIGRATION_COMMAND_ERROR'
+    UNTAR_FAIL = 'PERF_MIGRATION_UNTAR_FAILED'
+    INVALID = 'PERF_MIGRATION_INVALID'
+    WARNING = 'PERF_MIGRATION_WARNING'
+    FAILURE = 'PERF_MIGRATION_FAILURE'
+    SUCCESS = 'PERF_MIGRATION_SUCCESS'
+
+
+class PerfDataImportError(ImportError):
+    def __init__(self, error_string, return_code):
+        super(PerfDataImportError, self).__init__(error_string, return_code)
 
 
 def init_command_parser(subparsers):
     perf_parser = subparsers.add_parser('perf', help='migrate performance data')
+    # TBD collect the rrd top and perf data top
+    perf_parser.add_argument('rrd_dir',
+                             help="Top directory for a 4.x rrd tree")
+    perf_parser.add_argument('-t', '--perf_top', dest='perf_top', default="",
+                             help="Parent directory of the rrd trees")
     return perf_parser
 
 
 class Migration(MigrationBase):
-    pass
+    def __init__(self, args, progressCallback):
+        # common setup
+        super(Migration, self).__init__(args, progressCallback)
+
+        # check and install the scripts for mariadb and opentsdb services
+        self.rrd_dir = os.path.abspath(args.rrd_dir)
+        self.perf_top = os.path.abspath(args.perf_top)
+        if not os.path.exists(_import4_vol):
+            print _import_prefix + "%s does not exist" % _import4_vol
+            raise PerfDataImportError(Results.RUNTIME_ERROR, -1)
+        # for the imp4mariadb and imp4opentsdb services
+        if (not os.path.exists(_import4_pkg) or
+           not os.path.exists(_import4_pkg_bin)):
+            # this call should not execute under '/import4/pkg'
+            # instead, via <zenpack_path>/bin/import4
+            _rc = subprocess.call(["%s/install_pkg.sh" % sys.path[0]],
```

Comment:
Will this get called to re-copy over new files if I upgrade the zenpack and then run this again?